libgnucash/engine/Account.cpp

```diff
@@ -4084,33 +4075,46 @@ xaccAccountForEachLot(const Account *acc,
 static void
 set_boolean_key (Account *acc, std::vector<std::string> const & path, gboolean option)
 {
-    GValue v = G_VALUE_INIT;
     g_return_if_fail(GNC_IS_ACCOUNT(acc));
 
-    g_value_init (&v, G_TYPE_BOOLEAN);
-    g_value_set_boolean (&v, option);
+    set_kvp_string_path (acc, path, option ? "true" : nullptr);
     xaccAccountBeginEdit (acc);
-    qof_instance_set_path_kvp (QOF_INSTANCE (acc), &v, path);
     mark_account (acc);
     xaccAccountCommitEdit (acc);
-    g_value_unset (&v);
 }
 
 static gboolean
 boolean_from_key (const Account *acc, std::vector<std::string> const & path)
 {
-    GValue v = G_VALUE_INIT;
-    gboolean retval = FALSE;
     g_return_val_if_fail(GNC_IS_ACCOUNT(acc), FALSE);
-    qof_instance_get_path_kvp (QOF_INSTANCE(acc), &v, path);
-    if (G_VALUE_HOLDS_INT64 (&v))
-        retval = (g_value_get_int64 (&v) != 0);
-    if (G_VALUE_HOLDS_BOOLEAN (&v))
-        retval = (g_value_get_boolean (&v));
-    if (G_VALUE_HOLDS_STRING (&v))
-        retval = !strcmp (g_value_get_string (&v), "true");
-    g_value_unset (&v);
-    return retval;
+
+    auto kval = QOF_INSTANCE (acc)->kvp_data->get_slot (path);
+
+    if (!kval)
+        return false;
+
+    if (kval->get_type () == KvpValue::Type::INT64)
+        return (kval->get<int64_t>() != 0);
+
+    if (kval->get_type () == KvpValue::Type::STRING)
+        return (!g_strcmp0 (kval->get<const char*>(), "true"));
+
```

Comment:
the old code has G_VALUE_HOLDS_BOOLEAN which doesn't seem to be a valid Kvp type so there's no equivalent code for g_value_get_boolean.

Reply:
Yes, `g_value_from_kvp_value` can't return a boolean GValue so that branch wouldn't be exercised. This is fine.